jcli/build.rs

```diff
@@ -0,0 +1,82 @@
+use std::env::consts::{ARCH, OS};
+use std::process::Command;
+
+#[cfg(debug_assertions)]
+const BUILD_TYPE: &'static str = "debug";
+#[cfg(not(debug_assertions))]
+const BUILD_TYPE: &'static str = "release";
+
+fn main() {
+    let version_string = format!(
```

Comment:
There's a problem of code duplication, the 2 `build.rs`s are literally a copy-paste.

I think that this could be solved by moving this string generation to a new macro in `jormungandr-lib`.

The only things that really must be retrieved in build.rs are git info, rustc version and `PROFILE`. All the rest can be fetched with `env!` macro in production code. All of git info, rustc version and `PROFILE` are the same across all crates, so it can be retrieved only once: in `jormungandr-lib`'s `build.rs`. It can be then stored in a constant. The macro can then use the global constant from `jormungandr-lib` and local crate properties fetched with `env!` making its output relevant wherever it's used.

What do you think about it, will it fly?

Reply:
> ... git info ... use the global constant from `jormungandr-lib` ...  will it fly?

Yes and No, since it depends on the build case scenarios.
Let's focus only on `git info`, since the other ones work fine in any case.

Works fine (global const contains correct `git info`) on:
 - Official/CI releases (since they build everything everytime)
 - Local first build
 - Local full rebuilds
 - Local incremental builds caused by changes that require `jormungandr-lib` to be rebuild
 - ...

My understanding is that in _local user (re)builds_:

1. if there are changes (pulled/created) that need `jormungandr-lib` to be rebuilt, `jormungandr`/`jcli` binaries will be always rebuilded - global const contains **correct/actual** `git info`
2. if there are changes (pulled/created) that need only `jormungandr`/`jcli` binary to be rebuild, `jormungandr-lib` will not be rebuilded by default - global const contains **incorrect/old** `git info`!!!

If my understanding is correct, (**2**) may be problematic/confusing, if not I apologize for my misunderstanding.

Reply:
Your understanding is 100% correct, that will be a problem. To remove this duplication we probably would have to create a new crate intended only for use in `build.rs`s exposing this functionality. To be honest this could be quite useful when published on `crates.io`, I couldn't find any tool like that. Let's keep it like that for now.